src/cose/decoder.rs

```diff
@@ -164,6 +197,7 @@ pub fn decode_signature(bytes: Vec<u8>, payload: &[u8]) -> Result<CoseSignatures
         certs: Vec::new(),
         to_verify: bytes_to_verify,
     };
+    let mut result = CoseSignatures { values: Vec::new() };
```

Comment:
We probably don't need CoseSignatures as a type. We could just return a `Vec<CoseSignature>`.